pkg/cloudprovider/aws/instancetype.go

```diff
@@ -41,14 +41,13 @@ func (i *InstanceType) Zones() []string {
 	return i.ZoneOptions
 }
 
-func (i *InstanceType) Architectures() []string {
-	architectures := []string{}
+func (i *InstanceType) Architecture() string {
 	for _, architecture := range i.ProcessorInfo.SupportedArchitectures {
 		if value, ok := v1alpha1.AWSToKubeArchitectures[aws.StringValue(architecture)]; ok {
-			architectures = append(architectures, value)
+			return value
 		}
 	}
-	return architectures
+	return aws.StringValue(i.ProcessorInfo.SupportedArchitectures[0]) // Unrecognized, but used for error printing
```

Comment:
should you just print both then if it's just used for error printing (join) ?  Might be confusing if for some reason the first arch is `i386`